From 67e47439c832b0bef7ad9ad36a16b757df80eccf Mon Sep 17 00:00:00 2001
Message-Id: <67e47439c832b0bef7ad9ad36a16b757df80eccf.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Ramalingam C <ramalingam.c@intel.com>
Date: Tue, 17 Feb 2015 17:36:45 +0530
Subject: [PATCH 06/28] MUST_REBASE [VPG]: drm/i915: Wrapper function for DRRS
 VBT Data

This patch retrieves the DRRS_support_type and also DRRS_min_vrefresh
parsed from vbt in i915 for use inside ADF driver.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5610
Change-Id: I25298d6c27d3cc0f64568f964ea3048eb133060c
Signed-off-by: Ramalingam C <ramalingam.c@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |   12 ++++++++++++
 include/drm/i915_adf.h                  |    2 ++
 2 files changed, 14 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 00e4868..11b1ff2 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -269,6 +269,18 @@ out:
 	return ret;
 }
 EXPORT_SYMBOL(i915_adf_simple_buffer_alloc);
+
+u32 intel_get_vbt_drrs_support(void)
+{
+	return i915_adf_dev->vbt.drrs_type;
+}
+EXPORT_SYMBOL(intel_get_vbt_drrs_support);
+
+u32 intel_get_vbt_drrs_min_vrefresh(void)
+{
+	return i915_adf_dev->vbt.drrs_min_vrefresh;
+}
+EXPORT_SYMBOL(intel_get_vbt_drrs_min_vrefresh);
 #else
 int intel_adf_context_on_event(void) { return 0; }
 
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index e0806ce..385085d 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -55,6 +55,8 @@ extern u16 intel_get_dsi_port_frm_vbt(void);
 extern void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num);
 extern struct i2c_adapter *intel_adf_get_gmbus_adapter(u8 port);
 extern unsigned short *intel_get_vbt_pps_delays(void);
+extern u32 intel_get_vbt_drrs_support(void);
+extern u32 intel_get_vbt_drrs_min_vrefresh(void);
 
 /* ADF register calls for audio driver */
 int adf_hdmi_audio_register(void *drv, void *had_data);
-- 
1.7.9.5

